fabric-item-api-v1/src/main/java/net/fabricmc/fabric/api/client/item/v1/UpdateAnimationHandler.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.client.item.v1;
+
+import net.minecraft.item.ItemStack;
+
+import net.fabricmc.api.EnvType;
+import net.fabricmc.api.Environment;
+
+/**
+ * Represents an {@link net.minecraft.item.Item} that decides whether to run its NBT update animation.
+ */
+@Environment(EnvType.CLIENT)
+public interface UpdateAnimationHandler {
```

Comment:
Can we register these separately from items, like how mojang handles item model properties? This allows mods to tweak stuff from outside of their mod as well. That way you can drop the client/server concern largely as well.